src/plugins/drawing/components/drawing-tile.tsx

```diff
@@ -154,6 +157,40 @@ const DrawingToolComponent: React.FC<IDrawingTileProps> = observer(function Draw
     };
   };
 
+  const handleNavigatorPan = (direction: NavigatorDirection) => {
+    const drawingContainer = drawingLayerViewRef.current?.viewRef.current;
+    const svgElement = drawingContainer?.firstChild as SVGSVGElement;
+    const drawingContainerWidth = tileElt?.getBoundingClientRect().width ?? 0;
+    const drawingContainerHeight = tileElt?.getBoundingClientRect().height ?? 0;
+    const svgElementWidth = svgElement?.getBoundingClientRect().width;
+    const svgElementHeight = svgElement?.getBoundingClientRect().height;
+    const currTranslateX = contentModel.offsetX;
+    const currTranslateY = contentModel.offsetY;
+    const moveStep = 50;
+    // the maximum allowed distance to move the svg element in each direction
+    const maxTranslateX = Math.max(0, (svgElementWidth - drawingContainerWidth) / 2);
+    const maxTranslateY = Math.max(0, (svgElementHeight - drawingContainerHeight) / 2);
```

Comment:
I'm trying to understand what these limits are intended to do.  It feels like I can keep clicking the pan buttons and nothing stops me.

Reply:
I believe I've fixed that to some degree and made the code a little clearer. The result isn't perfect, unfortunately, but does limit how far you can pan to something reasonable.